tests/AbstractEditableRepositoryTest.php

```diff
@@ -488,4 +492,31 @@ public function testLinkTargetWithoutRepository()
         $link = new LinkResource('/webmozart/file');
         $link->getTarget();
     }
+
+    public function testChangeStreamGetStack()
+    {
+        $stream = new InMemoryChangeStream();
+
+        $this->writeRepo = $this->createWriteRepository($stream);
+        $this->writeRepo->add('/path', $v1 = new FileResource(__DIR__.'/Fixtures/dir1/file1'));
+        $this->writeRepo->add('/path', $v2 = new FileResource(__DIR__.'/Fixtures/dir1/file2'));
+        $this->writeRepo->add('/path', $v3 = new FileResource(__DIR__.'/Fixtures/dir2/file2'));
+
+        $resourceStack = $this->writeRepo->getStack('/path');
+
+        $this->assertInstanceOf('Puli\Repository\Api\ChangeStream\ResourceStack', $resourceStack);
+        $this->assertCount(3, $resourceStack->getAvailableVersions());
+        $this->assertFileResourcesEquals($v1, $resourceStack->getFirstVersion());
```

Comment:
It's just that the created resource is not associated to the repository so it doesn't have a path. However, on call on `getStack`, the resources found are associated to the current repo and path. So the objects and not exactly the same.